### PR TITLE
feat: Allowing traffic only through ALB for Rstudio

### PR DESCRIPTION
**Pull request by manikandan-thangavelu-rl**
Issue #, if available:

Description of changes:

Incorporated the PR comments in previous PRs
Added logic to allow only traffic from ALB to Rstudio instance

Checklist:

- [x] Have you successfully deployed to an AWS account with your changes?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully tested with your changes locally?

AS review ticket id:

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-cidr-service.test.js [hunk 1]
```diff
@@ -869,4 +869,49 @@ describe('EnvironmentScCidrService', () => {
...
     });
   });
+  describe('authorizeIngressRuleWithSecurityGroup', () => {
+    it('should call authorize security group', async () => {
+      service.authorizeSecurityGroupIngress = jest.fn();
+      service.getEc2Client = jest.fn(() => {
+        return {};
+      });
+      await service.authorizeIngressRuleWithSecurityGroup({}, 'envId', {}, 'groupId');
+      expect(service.authorizeSecurityGroupIngress).toHaveBeenCalled();
```
Comment (jn1119) [inline addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-cidr-service.test.js]: Verify that `authorizeSecurityGroupIngress` was called with correct arguments.

File: addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-cidr-service.test.js [hunk 2]
```diff
@@ -869,4 +869,49 @@ describe('EnvironmentScCidrService', () => {
...
+  });
+
+  describe('revokeIngressRuleWithSecurityGroup', () => {
+    it('should call revoke security group', async () => {
+      service.revokeSecurityGroupIngress = jest.fn();
+      service.getEc2Client = jest.fn(() => {
+        return {};
+      });
+      await service.revokeIngressRuleWithSecurityGroup({}, 'envId', {}, 'groupId');
+      expect(service.revokeSecurityGroupIngress).toHaveBeenCalled();
```
Comment (jn1119) [inline addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-cidr-service.test.js]: Verify that `revokeSecurityGroupIngress` was called with correct arguments.

File: addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/terminate-launch-dependency/terminate-launch-dependency.js [hunk 3]
```diff
@@ -116,6 +117,28 @@ class TerminateLaunchDependency extends StepBase {
...
+            envId,
+            updateRule,
+            instanceSecurityGroup,
+          );
+        } catch (error) {
+          // Don't fail the termination if revoke fails
+          this.print({
+            msg: `Security group rule revoke failed with error - ${error.message}`,
+          });
+        }
```
Comment (jn1119) [inline addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/terminate-launch-dependency/terminate-launch-dependency.js:141]: Didn't see a corresponding unit test change for this.

File: main/solution/machine-images/config/infra/files/rstudio/nginx.conf [hunk 4]
```diff
@@ -65,20 +65,12 @@ http {
...
-    }
-
     location / {
       proxy_pass http://localhost:8787;
-      proxy_redirect https://localhost:8787/ $scheme://$http_host/;
     }
 
     location /auth-do-sign-in {
       proxy_pass http://localhost:8787;
-      proxy_redirect https://localhost:8787/ $scheme://$http_host/;
```
Comment (jn1119) [inline main/solution/machine-images/config/infra/files/rstudio/nginx.conf:81]: Why was this removed?

File: main/solution/machine-images/config/infra/files/rstudio/nginx.conf [hunk 5]
```diff
@@ -65,20 +65,12 @@ http {
     proxy_set_header X-Real-IP $remote_addr;
     proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
 
-   # Health check url
-    location /health {
-      return 200 'OK';
-      add_header Content-Type text/plain;
-    }
-
```
Comment (jn1119) [inline main/solution/machine-images/config/infra/files/rstudio/nginx.conf:73]: why was this removed?

File: addons/addon-environment-sc-api/packages/environment-type-mgmt-services/lib/environment-type/env-type-candidate-service.js [hunk 6]
```diff
@@ -255,5 +270,25 @@ class EnvTypeCandidateService extends Service {
       ...args,
     );
   }
+
+  /**
+   * A private utility method to find a portfolio id from the department store table
+   * @returns {Promise<string>}
+   */
+  async getPortfolioId() {
```
Comment (jn1119) [inline addons/addon-environment-sc-api/packages/environment-type-mgmt-services/lib/environment-type/env-type-candidate-service.js:278]: Add a test?

File: addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-cidr-service.test.js (hunk 1 above)
Comment (manikandan-thangavelu-rl) [reply addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-cidr-service.test.js]: Incorporated the suggestion and updated the PR

File: addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-cidr-service.test.js (hunk 2 above)
Comment (manikandan-thangavelu-rl) [reply addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/__tests__/environment-sc-cidr-service.test.js]: Incorporated the suggestion and updated the PR

File: addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/terminate-launch-dependency/terminate-launch-dependency.js (hunk 3 above)
Comment (manikandan-thangavelu-rl) [reply addons/addon-environment-sc-api/packages/environment-sc-workflow-steps/lib/steps/terminate-launch-dependency/terminate-launch-dependency.js:141]: Incorporated the suggestion and updated the PR

File: main/solution/machine-images/config/infra/files/rstudio/nginx.conf (hunk 4 above)
Comment (manikandan-thangavelu-rl) [reply main/solution/machine-images/config/infra/files/rstudio/nginx.conf:81]: These are changes needed for RstudioV2. Since we are going to have RstudioV2 in RL repository, reverted this file back to original version.

File: main/solution/machine-images/config/infra/files/rstudio/nginx.conf (hunk 5 above)
Comment (manikandan-thangavelu-rl) [reply main/solution/machine-images/config/infra/files/rstudio/nginx.conf:73]: These are changes needed for RstudioV2. Since we are going to have RstudioV2 in RL repository, reverted this file back to original version.

File: addons/addon-environment-sc-api/packages/environment-type-mgmt-services/lib/environment-type/env-type-candidate-service.js (hunk 6 above)
Comment (manikandan-thangavelu-rl) [reply addons/addon-environment-sc-api/packages/environment-type-mgmt-services/lib/environment-type/env-type-candidate-service.js:278]: Incorporated the suggestion and updated the PR

**Review by jn1119 — APPROVED** [review]
